pkg_resources/__init__.py

```diff
@@ -414,10 +414,9 @@ def _macos_vers(_cache=[]):
         if version == '':
             plist = '/System/Library/CoreServices/SystemVersion.plist'
             if os.path.exists(plist):
-                if hasattr(plistlib, 'readPlist'):
-                    plist_content = plistlib.readPlist(plist)
-                    if 'ProductVersion' in plist_content:
-                        version = plist_content['ProductVersion']
+                plist_content = plistlib.load(plist)
```

Comment:
My wrong. I don't have a Mac so cannot really test this. Perhaps it would be worth creating a unit test for this one.

Reply:
Yes I agree, it is very tricky to test (I also don't have macOS).

Reply:
In this PR? I would rather defer to another PR.

I would need to find a Mac to extract a `/System/Library/CoreServices/SystemVersion.plist`. Can we do that for testing purposes without infringing Apple's copyright?

Reply:
The way I am looking into it is a regression test, so it would make sense to add it as the very first commit of the PR.

> I would need to find a Mac to extract a /System/Library/CoreServices/SystemVersion.plist. Can we do that for testing purposes without infringing Apple's copyright?

I think we can mock it based on the code that reads it, instead of copying an existing file, so there should be no problems in terms of copyright (because we basically are guessing how the content would look like).
@DimitriPapadopoulos, if that is OK with you I will prepare something and add it to this PR.

Reply:
Sure, please do. That would be a good addition.

Reply:
Perfect, I added 2fdfcb9056aa202544e233f3da5dbce4f7cb1af6 before the change and tested it with Python 3.8.

After the change I removed the `skipif` and the warning ignore in d8307f32df7558768b4f889a82472e269e6152c9, testing it again with Python 3.11.

It is not the most beautiful and perfect test in the word, but I think it will do as a regression test...

Reply:
Please let me know if you are OK with the PR as it is, then we can merge.

Reply:
Looks good. Will you squash the commits?

Reply:
I can if you prefer. Maybe also combine them in a meaningful way.